pennylane/gradients/spsa_gradient.py

```diff
@@ -302,6 +305,17 @@ def spsa_grad(
         # Skip the unshifted tape
         shifts = shifts[1:]
 
+    if sampler_rng is None:
+        sampler_rng = np.random.default_rng()
+    elif isinstance(sampler_rng, int):
+        sampler_rng = np.random.default_rng(sampler_rng)
+    elif not isinstance(sampler_rng, np.random.Generator):
+        err = (
+            'Argument sampler_rng has invalid type. Expected an integer, '
+            f'or np.random.Generator, or None. Received {type(sampler_rng)}'
+        )
+        raise ValueError(err)
```

Comment:
```suggestion
    sampler_rng = np.random.default_rng(sampler_rng)
```

See documentation of `default_rng`:

> If passed a [Generator](https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.Generator), it will be returned unaltered.

The initialization logic here can just be handled by `default_rng`.

Reply:
Oh, nice!